docs/app/components/sample-data.js

```diff
@@ -1,27 +1,42 @@
 import Component from '@glimmer/component';
-import { cached } from '@glimmer/tracking';
 import { setComponentTemplate } from '@ember/component';
 import { hbs } from 'ember-cli-htmlbars';
 
 import { generateSampleData } from 'docs/data';
+import { task, timeout } from 'ember-concurrency';
+import { taskFor } from 'ember-concurrency-ts';
+import { modifier } from 'ember-modifier';
 
 export default class SampleData extends Component {
-  @cached
-  get data() {
+  setup = modifier(() => {
+    taskFor(this._getData).perform();
+  });
+
+  @task
+  *_getData() {
+    const wait = this.args.timeout ?? 0;
+    // Simulate waiting for an async request
+    yield timeout(wait);
+
     return generateSampleData(this.args.rows, this.args.columns);
   }
 
+  get data() {
+    return taskFor(this._getData).lastSuccessful?.value;
+  }
+
   get columns() {
-    return this.data.headers;
+    return this.data?.headers;
   }
 
   get rows() {
-    return this.data.rows;
+    return this.data?.rows;
   }
 }
 
 setComponentTemplate(
   hbs`
+    {{this.setup}}
```

Comment:
this should break, I think, because a modifier is being applied to non-element space.

Instead, do you want to use `trackedFunction` from ember-resources?

https://ember-resources.pages.dev/functions/util_function.trackedFunction